doc/source/io.rst

```diff
@@ -4056,31 +4056,58 @@ Compression
 +++++++++++
 
 ``PyTables`` allows the stored data to be compressed. This applies to
-all kinds of stores, not just tables.
+all kinds of stores, not just tables. Two parameters are used to
+control compression: ``complevel`` and ``complib``.
+
+``complevel`` specifies if and how hard data is to be compressed.
+              ``complevel=0`` and ``complevel=None`` disables
+              compression and ``0<complevel<10`` enables compression.
+              
+``complib`` specifies which compression library to use. If nothing is
+            specified the default library ``zlib`` is used. A
+            compression library usually optimizes for either good
+            compression rates or speed and the results will depend on
+            the type of data. Which type of
+            compression to choose depends on your specific needs and
+            data. The list of supported compression libraries:
+
+             - ``zlib`` The default compression library. A classic in terms of compression, achieves good compression rates but is somewhat slow.
+             - ``lzo`` Fast compression and decompression.
+             - ``bzip2`` Good compression rates.
+             - ``blosc``: Fast compression and decompression.
+
+             .. versionadded:: 0.20.2
+                               
+                Support for alternative blosc compressors:
+                  
+                - ``blosc:blosclz`` This is the default compressor for ``blosc``
+                - ``blosc:lz4``
+                - ``blosc:lz4hc``  
+                - ``blosc:snappy``
+                - ``blosc:zlib``
+                - ``blosc:zstd``
+               
+             If ``complib`` is defined as something other than the
+             listed libraries a ``ValueError`` exception is issued.
 
-- Pass ``complevel=int`` for a compression level (1-9, with 0 being no
-  compression, and the default)
-- Pass ``complib=lib`` where lib is any of ``zlib, bzip2, lzo, blosc`` for
-  whichever compression library you prefer.
+.. note::
 
-``HDFStore`` will use the file based compression scheme if no overriding
-``complib`` or ``complevel`` options are provided. ``blosc`` offers very
-fast compression, and is my most used. Note that ``lzo`` and ``bzip2``
-may not be installed (by Python) by default.
+   ``lzo`` and ``bzip2`` may not be available on all platforms by
+   default. If the library is missing, compression defaults to
```

Comment:
so, bzip2 is actually available on all platforms? Then I would say bzip2 should be removed from that note. I can make it more general and say that if the chosen library is missing (but the name checks out) compression defaults to zlib or I can remove it completely. 

There is no warning when a library is missing and compression defaults to zlib. 

I did the testing of the various libraries and I had to add a specialcase and remove ``lzo`` from some travis auto-tests or they would fail. ``bzip2`` was available on all the platforms.